src/context.js

```diff
@@ -1,10 +1,18 @@
 "use strict";
 module.exports = function(Promise) {
+
+var util = require("./util");
 var longStackTraces = false;
 var contextStack = [];
 
-Promise.prototype._promiseCreated = function() {};
-Promise.prototype._pushContext = function() {};
+util.wrapMethodIfExistsDefineIfNot(Promise.prototype, "_promiseCreated",
```

Comment:
Did you mean longStackTraces instead of cancellation? I could not find places in code where cancellation overrides any hooks. Could you point to it? Correction commit will contain new implementation of hooks wrapper that supports both "hooking" and "unhooking", so this problem will be handled.